sct-commons/src/main/java/org/lfenergy/compas/sct/commons/util/FcdaCsvHelper.java

```diff
@@ -5,41 +5,54 @@
 package org.lfenergy.compas.sct.commons.util;
 
 import com.opencsv.bean.CsvBindByPosition;
-import lombok.Getter;
+import lombok.*;
+import org.lfenergy.compas.scl2007b4.model.SCL;
 import org.lfenergy.compas.scl2007b4.model.TFCDA;
 import org.lfenergy.compas.scl2007b4.model.TFCEnum;
 
 import java.io.Reader;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 /**
- * This class is a helper method to load FCDA from a CSV file for use with {@link IHmiService#createAllHmiReportControlBlocks}
+ * This class is a helper method to load FCDA from a CSV files for use with
+ * {@link org.lfenergy.compas.sct.commons.HmiService#createAllHmiReportControlBlocks(SCL, List)}
+ * {@link org.lfenergy.compas.sct.commons.ExtRefService#createDataSetAndControlBlocks(SCL, Set)}
+ * {@link org.lfenergy.compas.sct.commons.ExtRefService#createDataSetAndControlBlocks(SCL, String, Set)}
+ * {@link org.lfenergy.compas.sct.commons.ExtRefService#createDataSetAndControlBlocks(SCL, String, String, Set)}
  * Use the getter to access the list of parsed FCDA.
  *
  * @see CsvUtils
- * @see IHmiService#createAllHmiReportControlBlocks
+ * @see org.lfenergy.compas.sct.commons.HmiService.createAllHmiReportControlBlocks
+ * @see org.lfenergy.compas.sct.commons.ExtRefService.createDataSetAndControlBlocks
  */
 public class FcdaCsvHelper {
 
     @Getter
-    private final List<TFCDA> fcdas;
+    private final List<TFCDA> fcdaForHmiReportControls;
+
+    @Getter
+    private final Set<FcdaForDataSets> fcdaForDataSets;
 
     /**
      * Constructor
-     * Provide the CSV file as a Reader. For example, you can create a reader like this :
+     * Provide the CSV files as a Reader. For example, you can create a reader like this :
      * <code>new InputStreamReader(getClass().getClassLoader().getResourceAsStream(fileName), StandardCharsets.UTF_8);</code>
      *
-     * @param csvSource a reader that provides the data as CSV
+     * @param csvSourceForHmiReportControl        a reader that provides the FCDA datas for HMI ReportControl Blocks as CSV
+     * @param csvSourceForDataSetAndControlBlocks a reader that provides the FCDA datas for DataSets and Control Blocks creation as CSV
      */
-    public FcdaCsvHelper(Reader csvSource) {
-        fcdas = CsvUtils.parseRows(csvSource, Row.class).stream()
-                .map(row ->
-                        SclConstructorHelper.newFcda(row.ldInst, row.lnClass, row.lnInst, row.prefix, row.doName, null, row.fc)
+    public FcdaCsvHelper(Reader csvSourceForHmiReportControl, Reader csvSourceForDataSetAndControlBlocks) {
+        fcdaForHmiReportControls = CsvUtils.parseRows(csvSourceForHmiReportControl, FcdaForHmiReportControl.class).stream()
+                .map(fcdaForHmiReportControl ->
```

Comment:
I still don't understand why we map a second time juste after OpenCsv...

Reply:
Because TFcda objects should be create and file is not mapped directly to jaxb objects